### PR TITLE
chore: improve v2 types

### PR DESCRIPTION
**Pull request by w3b6x9**
## What kind of change does this PR introduce?

Chore

## What is the new behavior?

Improve and export Realtime types for v2 release.

### CODE REVIEW

File: src/RealtimeChannel.ts [hunk 1]
```diff
@@ -253,78 +277,49 @@ export default class RealtimeChannel {
...
+      filter?: string
+    },
+    callback: (payload: RealtimePostgresChangesPayload<T>) => void
+  ): RealtimeChannel
+  on(
+    type: `${REALTIME_LISTEN_TYPES}`,
+    filter: { event: string; [key: string]: string },
+    callback: (payload: any) => void
+  ): RealtimeChannel {
+    return this._on(type, filter, callback)
```
Comment (dshukertjr) [inline src/RealtimeChannel.ts:314]: Love this! It's such a massive DX improvement!

**Review by dshukertjr — COMMENTED** [review]
Love the new strongly typed realtime-js so much!

**Review by alaister — APPROVED** [review]
lgtm! Thanks Wen Bo!

**Comment by w3b6x9** [comment]
@alaister @dshukertjr thanks for all your help!

**Comment by ruggi99** [comment]
Hi.
Is there a reason for removing `onOpen`, `onClose`, `onError`, and `onMessage` functions in `RealtimeClient.ts`?

Personally `onOpen` was useful.
If I have a reconnection of the websocket connection I'd like to be informed in order to fetch (probably) stale data.
I didn't see any new implementation, how can I achieve again this goal?

Thanks

**Pull request merged**